src/core/ledger/testUtils.js

```diff
@@ -1,12 +1,23 @@
 // @flow
+
+import cloneDeep from "lodash.clonedeep";
+import {Ledger} from "./ledger";
+import {newIdentity, type Identity, type IdentityId} from "../identity";
+import * as G from "./grain";
 import * as uuid from "../../util/uuid";
 
 export interface UuidMock {
   resetFakeUuid(): void;
   setNextUuid(id: uuid.Uuid): void;
 }
 
-export function createUuidMock(): UuidMock {
+export interface DateMock {
+  resetFakeDate(): void;
+  setFakeDate(id: number): void;
+}
+
+let uuidMock = _createUuidMock();
```

Comment:
Seems like we're still leaking test-mutable state into the module scope here.

Reply:
fair enough... Looks like to do it right I'll need to move the identity and ledger builders into the mock state. Shouldn't be too hard.